p2p/switch_test.go

```diff
@@ -477,6 +476,67 @@ func TestSwitchFullConnectivity(t *testing.T) {
 	}
 }
 
+func TestSwitchAcceptRoutine(t *testing.T) {
+	cfg.MaxNumInboundPeers = 5
+
+	// make switch
+	sw := MakeSwitch(cfg, 1, "testing", "123.123.123", initSwitchFunc)
+	err := sw.Start()
+	require.NoError(t, err)
+	defer sw.Stop()
+
+	remotePeers := make([]*remotePeer, 0)
+
+	// 1. check we connect up to MaxNumInboundPeers
+	for i := 0; i < cfg.MaxNumInboundPeers; i++ {
+		rp := &remotePeer{PrivKey: ed25519.GenPrivKey(), Config: cfg}
+		remotePeers = append(remotePeers, rp)
+		rp.Start()
+		_, err = rp.Dial(sw.NodeInfo().NetAddress())
+		require.NoError(t, err)
+	}
+	assertNPeersWithTimeout(t, sw, cfg.MaxNumInboundPeers, 250*time.Millisecond)
+
+	// 2. check we close new connections if we already have MaxNumInboundPeers peers
+	rp := &remotePeer{PrivKey: ed25519.GenPrivKey(), Config: cfg}
+	rp.Start()
+	conn, err := rp.Dial(sw.NodeInfo().NetAddress())
+	require.NoError(t, err)
+	// check conn is closed
+	one := make([]byte, 1)
+	conn.SetReadDeadline(time.Now().Add(10 * time.Millisecond))
+	_, err = conn.Read(one)
+	if assert.Error(t, err) {
```

Comment:
Should this fail without the fix?

Reply:
I tried commenting out the fix and it didn't fail. Not sure if I need to try something else, I just commented out the one Cleanup call in the max inbound peer handling.

Reply:
```
--- FAIL: TestSwitchAcceptRoutine (0.04s)
        Error Trace:    switch_test.go:511
        Error:          Not equal:
                        expected: *errors.errorString(&errors.errorString{s:"EOF"})
                        actual  : *net.OpError(&net.OpError{Op:"read", Net:"tcp", Source:(*net.TCPAddr)(0xc00022c960), Addr:(*net.TCPAddr)(0xc00022c990), Err:(*poll.TimeoutError)(0xfbdec0)})
        Test:           TestSwitchAcceptRoutine
FAIL
exit status 1
```

fails for me